WIP: Build openmicroscopy using new build jars from repository

manics opened the pull request:
I've no idea why this doesn't work with just one pom.xml.

Example: `docker build -t openmicroscopy --build-arg MAVEN_SNAPSHOTS_REPO_URL=http://internal.host/nexus/repository/maven-internal/ .`

manics committed:
m3 -> SNAPSHOT

This reverts commit 90d2f3f8aa5acc1d6d55eb482db10301be6c5905.

manics committed:
Add pom.yml to download snapshot jars

manics committed:
Add omero-blitz python to pom.xml

manics committed:
Need two separate poms to get all required dependencies

Using just one pom leaves some unresolved dependencies. Using two seems to work. Use this to download the jars before running ./build.py:

    mvn -f download-repo-jars/pom1.xml dependency:copy-dependencies
    mvn -f download-repo-jars/pom2.xml dependency:copy-dependencies

manics committed:
Update Dockerfile to use remote repo deps

joshmoore commented:
Not sure if this has been tried, but something similar to:

```
git grep local-maven2
etc/ivysettings.xml:  <property name="local-maven2-dir" value="${user.home}/.m2/repository/"
```

might work here based on the "feature" of ant that once a property has been set it won't be reset, i.e.

 * add a property pointing to some (usually) non-extant maven server (`local-maven-url-=http://localhost:8081`)
 * add a resolver that is typically not used like `local-maven-server-resolver` which chains on top of ome-resolver
 * override ome.resolver to point to local-maven-server-resolver (in `*.properties` or on the command-line)
 * override local-maven-url to point to nexus (in `*.properties` or on the command-line)

manics committed:
Add Jenkinsfile

manics committed:
Add optional BUILD_PY_ARGS to docs/hudson/OMERO.sh

manics committed:
Switch back to testintegration node for now

manics committed:
Fix typo in Jenkinsfile build.py arg

manics committed:
Use Jenkinsfile virtualenv to upgrade setuptools

Building OMERO requires a different version of setuptools for some strange reason.

manics committed:
Revert "Add optional BUILD_PY_ARGS to docs/hudson/OMERO.sh"

This reverts commit 6eed9e3a2e640758d0a1583848b83be66063ea8a.

manics committed:
Try using default ~/.m2 on testintegration to workaround missing generated java classes

manics committed:
Use shared .m2 from preceding jobs

manics commented:
I think this is a lost cause
